zebra-consensus/src/transaction/check.rs

```diff
@@ -135,3 +135,55 @@ pub fn disabled_add_to_sprout_pool(
 
     Ok(())
 }
+
+/// Check if a transaction has any internal spend conflicts.
+///
+/// An internal spend conflict happens if the transaction spends a UTXO more than once or if it
+/// reveals a nullifier more than once.
+///
+/// Consensus rule:
+///
```

Comment:
Missing transparent consensus rule:
```suggestion
/// Consensus rules:
///
/// "each output of a particular transaction
/// can only be used as an input once in the block chain.
/// Any subsequent reference is a forbidden double spend-
/// an attempt to spend the same satoshis twice."
///
/// https://developer.bitcoin.org/devguide/block_chain.html#introduction
///
```